jax/_src/nn/functions.py

```diff
@@ -1008,9 +1032,20 @@ def _check_shape_and_dtype(t: Array | None, shape: Sequence[int],
         mask_type = MaskType.CAUSAL
       elif query_seq_lengths is not None:
         mask_type = MaskType.PADDING
+      # CuDNN supports only the left window with an exclusive boundary with
+      # causal mask is enabled.
+      sliding_window = None
+      if local_window_size is not None:
+        l_window, r_window = local_window_size
+        if r_window == 0 or mask_type == MaskType.CAUSAL:
+          sliding_window = l_window + 1
+        else:
+          raise ValueError(f"cuDNN doesn't support right window: {r_window}")
```

Comment:
This is not specific enough. I think cuDNN works with right window as long as causal attention is not used?

Reply:
Not really. cudnn requires causal mask and only works for l_window. Added more text in the error message.